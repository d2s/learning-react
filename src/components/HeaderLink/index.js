import React from 'react'
import classNames from 'classnames'

/**
 * CSS class names
 */
const cx = classNames('link dim gray f6 f5-ns dib mr3')
// const cx = classNames('link dim gray f6 f5-ns dib mr3', {
//   'black b': isActive
// })

export default (props) => (
  <a className={cx} title={props.name} href="#">
    {props.name}
  </a>
)
