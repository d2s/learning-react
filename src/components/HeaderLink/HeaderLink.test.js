import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

import HeaderLink from './index';

describe('With ReactDOM', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HeaderLink />, div);
  });
})

describe('With Enzyme', () => {
  it('shallow renders without crashing', () => {
    shallow(<HeaderLink />);
  });
})

describe('With Snapshot Testing', () => {
  it('HeaderLink renders the same way', () => {
    const component = renderer.create(<HeaderLink />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
