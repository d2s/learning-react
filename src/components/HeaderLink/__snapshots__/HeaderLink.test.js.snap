// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing HeaderLink renders the same way 1`] = `
<a
  className="link dim gray f6 f5-ns dib mr3"
  href="#"
/>
`;
